File: sdk/runanywhere-flutter/lib/data/network/models/auth/refresh_token_request.dart
/// Request model for token refresh.
///
/// Matches iOS `RefreshTokenRequest` from RunAnywhere SDK.
class RefreshTokenRequest {
  final String deviceId;
  final String refreshToken;

  const RefreshTokenRequest({
    required this.deviceId,
    required this.refreshToken,
  });

  Map<String, dynamic> toJson() => {
        'device_id': deviceId,
        'refresh_token': refreshToken,
      };
}

File: sdk/runanywhere-swift/Sources/RunAnywhere/Data/Network/Models/Auth/RefreshTokenRequest.swift
import Foundation

/// Request model for token refresh
public struct RefreshTokenRequest: Codable, Sendable {
    public let deviceId: String
    public let refreshToken: String

    public init(deviceId: String, refreshToken: String) {
        self.deviceId = deviceId
        self.refreshToken = refreshToken
    }

    enum CodingKeys: String, CodingKey {
        case deviceId = "device_id"
        case refreshToken = "refresh_token"
    }
}
